scripts/scil_compute_bundle_profiles.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Compute bundle profiles and their statistics along streamlines.
+"""
+
+
+import argparse
+import json
+import os
+
+import nibabel as nib
+import numpy as np
+
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.image import assert_same_resolution
+from scilpy.io.utils import (assert_inputs_exist,
+                             add_json_args,
+                             add_overwrite_arg,
+                             add_reference_arg)
+from scilpy.utils.filenames import split_name_with_nii
+from scilpy.utils.metrics_tools import get_bundle_metrics_profiles
+from scilpy.tracking.tools import resample_streamlines_num_points
+from scilpy.tractanalysis.features import get_streamlines_centroid
+
+
+def norm_l2(x):
+    return np.sqrt(np.sum(np.power(x, 2), axis=1, dtype=np.float32))
+
+
+def average_euclidean(x, y):
+    return np.mean(norm_l2(x - y))
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file to compute the bundle profiles on.')
+    p.add_argument('in_metrics', nargs='+',
+                   help='Nifti metric(s) on which to compute '
+                        'the bundle profiles.')
+
+    g = p.add_mutually_exclusive_group()
+    g.add_argument('--in_centroid',
+                   help='If provided it will be used to make sure all '
+                        'streamlines go in the same direction. \nOtherwise, a '
+                        'centroid will be automatically computed.\n'
+                        'Number of point per streamline will be set according'
+                        ' to centroid.')
+    g.add_argument('--nb_pts_per_streamline',
+                   type=int, default=20,
+                   help='Subsample each streamline to this number of points '
+                        '[%(default)s].')
+
+    add_json_args(p)
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_bundle] + args.in_metrics,
+                        optional=args.in_centroid)
+
+    if args.nb_pts_per_streamline <= 1:
+        parser.error('--nb_pts_per_streamline {} needs to be greater than '
+                     '1'.format(args.nb_pts_per_streamline))
+
+    assert_same_resolution(args.in_metrics)
+    metrics = [nib.load(m) for m in args.in_metrics]
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+
+    bundle_name, _ = os.path.splitext(os.path.basename(args.in_bundle))
+    stats = {}
+    if len(sft) == 0:
```

Comment:
We should have an explicit agreement on that.
I know some people like `if len(sft.streamlines) == 0` while I personally use `if len(sft)` and here you are using a mix of both. I don't care which one we choose.